frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.tsx

```diff
@@ -0,0 +1,84 @@
+import { t } from "ttag";
+
+import { Button, Flex, Icon, Select, TextInput } from "metabase/ui";
+import type * as Lib from "metabase-lib";
+
+import type { ColumnAndSeparator, ColumnOption } from "../../types";
+import { fromSelectValue, toSelectValue } from "../../utils";
+
+import styles from "./ColumnAndSeparatorRow.module.css";
+
+interface Props {
+  column: Lib.ColumnMetadata;
+  index: number;
+  options: ColumnOption[];
+  separator: string;
+  showLabels: boolean;
+  showRemove: boolean;
+  showSeparator: boolean;
+  onChange: (index: number, change: Partial<ColumnAndSeparator>) => void;
+  onRemove: (index: number) => void;
+}
+
+export const ColumnAndSeparatorRow = ({
+  column,
+  index,
+  options,
+  separator,
+  showLabels,
+  showRemove,
+  showSeparator,
+  onChange,
+  onRemove,
+}: Props) => {
+  return (
+    <Flex align="flex-end" gap={12}>
+      {showSeparator && (
+        <TextInput
+          className={styles.separator}
+          label={showLabels ? t`Separator` : undefined}
+          value={separator}
+          w={110}
+          onChange={event => {
+            const separator = event.target.value;
+            onChange(index, { separator });
+          }}
+        />
+      )}
+
+      <Select
+        className={styles.column}
+        data={options}
+        label={showLabels ? t`Column` : undefined}
+        styles={{
+          wrapper: {
+            "&:not(:only-child)": {
+              marginTop: 0,
+            },
```

Comment:
We have `marginTop` defined for most Mantine inputs by default.
I think it's because of the assumption that if the input is `:not(:only-child)` there will be a label above it.
This assumption does not hold for `Select` because it never is the only child due to `<input type="hidden"` being rendered before it (see [screenshot](https://github.com/metabase/metabase/assets/6830683/ba484c9a-99fe-400b-9823-d20984b195fd)).

It'd be better to assume that every label that is `:not(:only-child)` will have an input underneath, and add `marginBottom` to labels instead. Though it'd be ideal if the low-level building blocks never made such assumptions at all.

Anyway, fixing that is out of the scope of this PR, so I added this band-aid here.